### PR TITLE
minor fixes to unit tests

### PR DESCRIPTION
**Pull request by gemal**

### CODE REVIEW

**Comment by markstory** [comment]
Looks good to me. Just waiting for travis to finish before merging.

**Comment by gemal** [comment]
`Fatal error: Call to undefined method SimpleXMLElement::key() in /home/travis/build/cakephp/cakephp/vendor/phpunit/phpunit/src/Framework/Constraint/Count.php on line 59`

is that something my pull caused?

**Comment by markstory** [comment]
Does the stacktrace include an assertion you changed?

File: tests/TestCase/Utility/XmlTest.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function testBuild()
         $xml = CORE_TESTS . 'Fixture/sample.xml';
         $obj = Xml::build($xml);
         $this->assertEquals('tags', $obj->getName());
-        $this->assertEquals(2, count($obj));
+        $this->assertCount(2, $obj);
```
Comment (antograssiot) [inline tests/TestCase/Utility/XmlTest.php]: This change causes error in the tests suite

File: tests/TestCase/Utility/XmlTest.php [hunk 2]
```diff
@@ -272,7 +272,7 @@ public function testFromArray()
         $obj = Xml::fromArray($xml, 'attributes');
         $this->assertTrue($obj instanceof \SimpleXMLElement);
         $this->assertEquals('tags', $obj->getName());
-        $this->assertEquals(2, count($obj));
+        $this->assertCount(2, $obj);
```
Comment (antograssiot) [inline tests/TestCase/Utility/XmlTest.php]: As well as this one but I didn't dig why...

**Review by antograssiot — CHANGES_REQUESTED** [review]
The errors in the tests suite are linked to some of your changes, see my comments.

File: tests/TestCase/Utility/XmlTest.php [hunk 3]
```diff
@@ -285,7 +285,7 @@ public function testFromArray()
         $obj = Xml::fromArray($xml);
         $this->assertTrue($obj instanceof \SimpleXMLElement);
         $this->assertEquals('tags', $obj->getName());
-        $this->assertCount(2, $obj);
+        $this->assertEquals(2, count($obj));
```
Comment (dereuromark) [inline tests/TestCase/Utility/XmlTest.php]: Ideally, assertSame() to assert the same type, even though that's pretty obvious here.

**Comment by gemal** [comment]
Now all the tests works :)

**Comment by antograssiot** [comment]
@gemal why does your diff now includes commits that are not yours and that were merged already ?

**Comment by dereuromark** [comment]
Rebasing an sqashing helps.

**Comment by gemal** [comment]
I have no idea. What command do I need to run?

**Comment by markstory** [comment]
@gemal You can run

```bash
git fetch upstream master
git rebase -i upstream/master
```

You may need to use a different name for `upstream` depending on what your remote names are.

**Comment by markstory** [comment]
I squash merged this into master. Thank you for the help @gemal 👍

**Pull request merged**